vault/logical_system_activation_flags_test.go

```diff
@@ -0,0 +1,492 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/enthelpers/activationflags"
+	"github.com/hashicorp/vault/helper/namespace"
+	"github.com/hashicorp/vault/helper/testhelpers/corehelpers"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/hashicorp/vault/sdk/physical"
+	"github.com/hashicorp/vault/sdk/physical/inmem"
+	"github.com/stretchr/testify/require"
+)
+
+// TestActivationFlags_Read tests the read operation for the activation flags.
+func TestActivationFlags_Read(t *testing.T) {
+	t.Run("given an initial state then read flags and expect all to be unactivated", func(t *testing.T) {
+		core, _, _ := TestCoreUnsealedWithConfig(t, &CoreConfig{})
+
+		resp, err := core.systemBackend.HandleRequest(
+			context.Background(),
+			&logical.Request{
+				Operation: logical.ReadOperation,
+				Path:      prefixActivationFlags,
+				Storage:   core.systemBarrierView,
+			},
+		)
+
+		require.NoError(t, err)
+		require.Equal(t, resp.Data, map[string]interface{}{
+			"activated":   []string{},
+			"unactivated": activationflags.NewFeatureActivationFlags().ValidActivationFeatures(),
+		})
+	})
+}
+
+// TestActivationFlags_BadFeatureName tests a nonexistent feature name or a missing feature name
+// in the activation-flags path API call.
+func TestActivationFlags_BadFeatureName(t *testing.T) {
+	core, _, _ := TestCoreUnsealedWithConfig(t, &CoreConfig{})
+
+	tests := map[string]struct {
+		featureName string
+	}{
+		"if no feature name is provided then expect unsupported path": {
+			featureName: "",
+		},
+		"if an invalid feature name is provided then expect unsupported path": {
+			featureName: "fake-feature",
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			resp, err := core.router.Route(
+				namespace.ContextWithNamespace(context.Background(), namespace.RootNamespace),
+				&logical.Request{
+					Operation: logical.UpdateOperation,
+					Path:      fmt.Sprintf("sys/%s/%s/%s", prefixActivationFlags, tt.featureName, verbActivationFlagsActivate),
+					Storage:   core.systemBarrierView,
+				},
+			)
+
+			require.Error(t, err)
+			require.Nil(t, resp)
+			require.Equal(t, err, logical.ErrUnsupportedPath)
+		})
+	}
+}
+
+// TestSystemBackend_activationFlagsPaths tests that the expected paths are returned based on the
+// HVD-related environment variables set and the administrative namespace.
+func TestSystemBackend_activationFlagsPaths(t *testing.T) {
```

Comment:
I think it would be ideal if the HVD behavior was limited to Enterprise repo wihch means most of these test cases moving to ent-only tests. Does that make sense to you or is there a reason it needs to stay here?